homeassistant/helpers/filter.py

```diff
@@ -0,0 +1,125 @@
+"""Deprecation helpers for Home Assistant."""
+import logging
+import inspect
+import statistics
+from collections import deque
+
+DEFAULT_WINDOW_SIZE = 5
+FILTER_LOWPASS = 1
+FILTER_OUTLIER = 2
+FILTERS = [FILTER_LOWPASS, FILTER_OUTLIER]
+
+
+class Filter(object):
+    """Filter outlier states, and smooth things out."""
+
+    logger = None
+
+    def __init__(self, filter_algorithm, window_size=DEFAULT_WINDOW_SIZE,
+                 **kwargs):
+        """Decorator constructor, selects algorithm and configures window.
+
+        Args:
+            filter_algorithm (int): must be one of the defined filters
+            window_size (int): size of the sliding window that holds previous
+                                values
+            kwargs (dict): arguments to be passed to the specific filter
+
+        """
+        module_name = inspect.getmodule(inspect.stack()[1][0]).__name__
+        Filter.logger = logging.getLogger(module_name)
+        Filter.logger.debug("Filter %s on %s", filter_algorithm, module_name)
+        self.filter = None
+        self.filter_args = kwargs
+        self.states = deque(maxlen=window_size)
+
+        if filter_algorithm in FILTERS:
+            if filter_algorithm == FILTER_LOWPASS:
```

Comment:
*Suggestion:* I think over time, lot's of different filters would be added, so a dictionary mapping would IMO make more sense.

Reply:
Actually thinking about using Enum, but this is a RFC :)

dict or enum ?

Reply:
> dict or enum ?

I haven't used `enum`s enough to make a sound judgement, but I think `dict` make config validation simpler.